pkg/index/tree.go

```diff
@@ -150,6 +176,10 @@ func nodesBetween[V Value](root *Node[V], from, to int, callback func(node V)) e
 			if child.IsText() {
 				toChild = to - pos
 			}
+			// pos can be affected by children traversal and callback call.
+			// so, update pos before these to keep it safe.
+			pos += child.PaddedLength()
```

Comment:
It is better to revert this because we can't be sure this approach protects the problem.